pystac/extensions/render.py

```diff
@@ -0,0 +1,428 @@
+    :class:`~pystac.Collection` or :class:`~pystac.Item` with
+    properties from the :stac-ext:`Render Extension <render>`. This class is
+    generic over the type of STAC Object to be extended (e.g. :class:`~pystac.Item`,
+    :class:`~pystac.Collection`).
+
+    To create a concrete instance of :class:`RenderExtension`, use the
+    :meth:`RenderExtension.ext` method. For example:
+
+    .. code-block:: python
+
+        >>> item: pystac.Item = ...
+        >>> xr_ext = RenderExtension.ext(item)
+
+    """
+
+    name: Literal["render"] = "render"
+
+    @classmethod
+    def get_schema_uri(cls) -> str:
+        return SCHEMA_URI_PATTERN.format(version=DEFAULT_VERSION)
+
+    @classmethod
+    def ext(cls, obj: T, add_if_missing: bool = False) -> RenderExtension[T]:
+        """Extend the given STAC Object with properties from the
+        :stac-ext:`Render Extension <render>`.
+
+        This extension can be applied to instances of :class:`~pystac.Collection`
+        or :class:`~pystac.Item`.
+
+        Raises:
+            pystac.ExtensionTypeError : If an invalid object type is passed.
+        """
+        if isinstance(obj, pystac.Collection):
+            cls.ensure_has_extension(obj, add_if_missing)
+            return CollectionRenderExtension(obj)
+        elif isinstance(obj, pystac.Item):
+            cls.ensure_has_extension(obj, add_if_missing)
+            return ItemRenderExtension(obj)
+        else:
+            raise pystac.ExtensionTypeError(
+                f"RenderExtension does not apply to type '{type(obj).__name__}'"
+            )
+
+    def apply(
+        self,
+        renders: dict[str, Render],
+    ) -> None:
+        """Applies the render extension fields to the extended
+        object.
+
+        Args:
+            renders: a dictionary mapping render names to
+            :class: `~pystac.extension.render.Render` objects.
+        """
+        self.renders = renders
+
+    @property
+    def renders(self) -> dict[str, Render]:
+        """A dictionary where each key is the name of a render and each
+        value is a :class:`~Render` object.
+        """
+        renders: dict[str, dict[str, Any]] = get_required(
+            self._get_property(RENDERS_PROP, dict[str, dict[str, Any]]),
+            self,
+            RENDERS_PROP,
+        )
+        return {k: Render(v) for k, v in renders.items()}
+
+    @renders.setter
+    def renders(self, v: dict[str, Render]) -> None:
+        self._set_property(
+            RENDERS_PROP,
+            map_opt(lambda renders: {k: r.to_dict() for k, r in renders.items()}, v),
+            pop_if_none=False,
+        )
+
+
+class CollectionRenderExtension(RenderExtension[pystac.Collection]):
+    """A concrete implementation of :class:`RenderExtension` on a
+    :class:`~pystac.Collection` that extends the properties of the Collection to include
+    properties defined in the :stac-ext:`Render Extension <xarray>`.
```

Comment:
Just caught a few errant references to `<xarray>` in the docstrings. Lemme push a fix.